woocommerce/woocommerce-ios
# [My Store updates] Update bar chart to line chart

Closes: #5737
Spike p2: p91TBi-6qN-p2


### Description


As part of design updates for the My Store tab, this PR updates the order stats chart from bar chart (`BarChartView`) to line chart (`LinChartView`) using the Charts library. I did a spike (p91TBi-6qN-p2) beforehand, and I spent some time on adding a shadow and border color to the highlighted circle and handling edge cases. The changes include:

- Updated bar chat `BarChartView` to line chart `LinChartView` in `StoreStatsV4PeriodViewController` nib
- Disabled circles for data points in the line chart (enabled by default, where each data point has a circle drawn)
- Updated the color and width of the data line and highlighted line
- Created a `Charts.MarkerImage` subclass for the circle marker `StoreStatsChartCircleMarker` shown when highlighting a data point (the pink circle in the screenshots above)
- Handled edge case when there is no revenue data: the line color is shown as clear, and the gradient is not configured
- Handled edge case when all revenue is negative: the gradient is not configured because the Charts library fills the area from the top of the chart instead of the baseline (y-axis value 0) and it looks odd from my testing


### Testing instructions


Because of the holidays, code reviewers feel free to point out anything deviating from the design that you notice. After most of the design updates are complete, I will ping the design team for design review and address any feedback.

- [x] @jaclync tests the charts when the feature flag is disabled

Prerequisite: the site ideally has no revenue for today, and some revenue for either this week, this month, or this year
- Launch the app
- If there is no revenue today, the Today tab should look like before
- If there is any revenue for this week/month/year, the chart should show a line chart instead of a bar chart
- Select a time in the chart --> a pink circle should be shown above the selected point in the chart

### Screenshots


mode \ scenario | no revenue | negative revenue only | positive revenue  | selecting a time
-- | -- | -- | -- | --
dark | ![Simulator Screen Shot - iPhone 11 - 2021-12-23 at 14 10 16](https://user-images.githubusercontent.com/1945542/147198613-de055276-3272-44a3-8a82-5a3d53b42e1f.png) | ![Simulator Screen Shot - iPhone 11 - 2021-12-23 at 14 10 32](https://user-images.githubusercontent.com/1945542/147198623-b687c859-fb5b-4f49-850c-015d9cbe3ba3.png) | ![Simulator Screen Shot - iPhone 11 - 2021-12-23 at 14 10 37](https://user-images.githubusercontent.com/1945542/147198627-c9704354-495c-4ac5-8cd5-4399b42fba89.png) | ![Simulator Screen Shot - iPhone 11 - 2021-12-23 at 14 10 44](https://user-images.githubusercontent.com/1945542/147198631-f2bca0dd-fb08-48a5-a14d-788dd685c72e.png)
light | ![Simulator Screen Shot - iPhone 11 - 2021-12-23 at 14 09 59](https://user-images.githubusercontent.com/1945542/147198753-48ce00f7-1131-4a05-bbe3-f87d8833037e.png)  | ![Simulator Screen Shot - iPhone 11 - 2021-12-23 at 14 09 05](https://user-images.githubusercontent.com/1945542/147198766-87d51087-4cb5-4a88-8d96-a83c4285b02d.png) | ![Simulator Screen Shot - iPhone 11 - 2021-12-23 at 14 09 10](https://user-images.githubusercontent.com/1945542/147198760-f82a57fb-c252-4f75-bae8-734750e6e32b.png) | ![Simulator Screen Shot - iPhone 11 - 2021-12-23 at 14 10 57](https://user-images.githubusercontent.com/1945542/147198768-776b37b2-0786-4deb-997b-9fa809fa8ce4.png)


---
